.travis.yml

```diff
@@ -1,58 +1,76 @@
 language:
-- java
-- python
+  - java
+  - python
+  - go
 notifications:
   email: false
   irc:
     template:
-    - "%{repository}#%{build_number} (%{branch} - %{commit} : %{author}): %{message}"
+      - "%{repository}#%{build_number} (%{branch} - %{commit} : %{author}): %{message}"
     channels:
-    - chat.freenode.net#enmasse
+      - chat.freenode.net#enmasse
     use_notice: true
     skip_join: true
 sudo: required
 services:
-- docker
+  - docker
 cache:
   directories:
-  - "$HOME/.m2"
+    - "$HOME/.m2"
 jdk:
 - openjdk8
 python:
-- "2.7.14"
+  - "2.7.14"
+go:
+  - "1.10.x"
 before_install:
-- sudo pip install requests[security] ansible
-- gem install asciidoctor
+  - sudo pip install requests[security] ansible
+  - gem install asciidoctor
+  - sudo pip install requests[security] ansible
+  - sudo add-apt-repository -y ppa:masterminds/glide && sudo apt-get update
+  - sudo apt-get install -y glide
+  - gem install asciidoctor
+  - gimme 1.10.5
+  - source ~/.gimme/envs/go1.10.5.env
+  - go version
 install: true # would only call "mvn install", which doesn't help us at this point
+before_script:
+  - pwd
+  - export GOPATH="$HOME/gopath"
```

Comment:
@ctron why we need these steps? also I don't understand why we create this path and them copy all enmasse conetent here?

Reply:
Welcome to the wonderful world of Go. Go has some specific requirements of its base path. And in this case I wasn't able to fix this with a symlink.

The "verify" step is needed in general to check if the code generator creates different files than the one that are currently checked in. If that is the case, then someone should manually run the code generator, and commit the changes.